app/multitenant/dynamo_collector.go

```diff
@@ -346,30 +341,50 @@ func (c *dynamoDBCollector) Add(ctx context.Context, rep report.Report) error {
 		return err
 	}
 	writer.Close()
-	bytes := buf.Bytes()
-	dynamoReportSize.WithLabelValues("PutItem").Add(float64(len(bytes)))
+	reportSize.Add(float64(buf.Len()))
 
+	// second, put the report on s3
 	now := time.Now()
 	rowKey := fmt.Sprintf("%s-%s", userid, strconv.FormatInt(now.UnixNano()/time.Hour.Nanoseconds(), 10))
-	startTime := time.Now()
-	resp, err := c.db.PutItem(&dynamodb.PutItemInput{
-		TableName: aws.String(c.tableName),
-		Item: map[string]*dynamodb.AttributeValue{
-			hourField: {
-				S: aws.String(rowKey),
-			},
-			tsField: {
-				N: aws.String(strconv.FormatInt(now.UnixNano(), 10)),
-			},
-			reportField: {
-				B: bytes,
+	colKey := strconv.FormatInt(now.UnixNano(), 10)
+	s3Key := fmt.Sprintf("%s/%s", rowKey, colKey)
+	err = timeRequest("Put", s3RequestDuration, func() error {
```

Comment:
Let's check that the rowKey gets hashed by s3, and that we don't need to do that, to ensure performance.